File: Solutions/1456. Maximum Number of Vowels in a Substring of Given Length/1456.java
class Solution {
    public int maxVowels(String s, int k) {
        int ans = 0, cur = 0;
        for (int i = 0; i < s.length(); i++) {
            if ( isVowel( s.charAt(i) ) ) {
                cur++;
            }
            if ( i >= k && isVowel( s.charAt(i-k) ) ) {
                cur--;
            }
            ans = Math.max(cur, ans);
        }
        return ans;
    }
    private boolean isVowel(Character c){
        return c  == 'a' || c  == 'e' || c  == 'i' || c  == 'o' || c  == 'u';
    }
}

File: Solutions/1456. Maximum Number of Vowels in a Substring of Given Length/1456.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        cur = 0
        ans = 0
        vowels = {'a', 'e', 'i', 'o', 'u'}
        for i, c in enumerate(s):
            if c in vowels:
                cur += 1
            if i>=k and s[i-k] in vowels:
                cur -= 1
            ans = max(cur, ans)
        return ans 